tasks/os-check.yml

```diff
@@ -3,3 +3,18 @@
   fail:
     msg: "The Datadog Ansible role does not support your OS yet. Please email support@datadoghq.com to open a feature request."
   when: ansible_facts.os_family not in ["RedHat", "Rocky", "AlmaLinux", "Debian", "Suse", "Windows", "Darwin"]
+
+- name: Upper bound last supported Agent version for Centos < 7
+  when: ansible_facts.os_family == "RedHat"
+  block:
+    - name: Get RHEL major version equivalent
+      command: "rpm -E %{rhel}"  # noqa: command-instead-of-module
+      register: rhel_version
+      changed_when: false
+      check_mode: false
+    - name: Set upper agent version bound as ansible fact
+      set_fact:
+        datadog_agent_max_version:
+          full: "{{ datadog_agent_major_version | default('7', true) }}.51.1"
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Do we want to make this work, even if we end up releasing another patch version?

Reply:
Not sure, I think it's better that we follow the behavior we set in the install script - but I don't want us to forget to bump this and leave a vulnerability sitting (for example). It's probably fine as-is, but if someone knows of a way to do `<.52` that feels better.

Reply:
I think specifying version range should work, but you'll have to use the epoch number (`1:`) which the Agent packages have had since the beginning.

I think something like `yum install "datadog-agent < 1:7.52.0-1"` should work, but you'd have to figure out how best to pass it to the ansible tasks which might be doing additional parsing etc.

Reply:
I'll give that a shot.

Reply:
This worked well 👍